cumulus/taskflow/core/__init__.py

```diff
@@ -0,0 +1,239 @@
+
+def create_ec2_cluster(task, cluster, profile, ami_spec):
+    machine_type = cluster['machine']['id']
+    nodeCount = cluster['clusterSize']-1
+    launch_spec = 'ec2'
+
+    # Look up the external IP of the deployment to user for firewall rules
+    r = requests.get(CHECKIP_URL)
+    r.raise_for_status()
+    source_ip = '%s/32' % r.text.strip()
+
+    extra_rules = [{
+        'proto': 'tcp',
+        'from_port': 9000,
+        'to_port': 9000,
+        'cidr_ip': source_ip
+    }]
+
+
+    task.logger.info('Using source ip: %s' % source_ip)
+
+    launch_params = {
+        'master_instance_type': machine_type,
+        'master_ami_spec': ami_spec,
+        'node_instance_count': nodeCount,
+        'node_instance_type': machine_type,
+        'node_ami_spec': ami_spec,
+        'gpu': cluster['machine']['gpu'],
+        'source_cidr_ip': source_ip,
+        'extra_rules': extra_rules
+    }
+    provision_spec = 'gridengine/site'
+    provision_params = {
+        'ansible_ssh_user': 'ubuntu'
+    }
+
+    body = {
+        'type': 'ec2',
+        'name': cluster['name'],
+         'config': {
+            'launch': {
+                'spec': launch_spec,
+                'params': launch_params
+            },
+            'provision': {
+                'spec': provision_spec
+            }
+        },
+        'profileId': cluster['profileId']
+    }
+    client = create_girder_client(
+        task.taskflow.girder_api_url, task.taskflow.girder_token)
+
+    try:
+        cluster = client.post('clusters',  data=json.dumps(body))
+    except HttpError as he:
+        task.logger.exception(he.responseText)
+        raise
+
+    msg = 'Created cluster: %s' % cluster['_id']
+    task.taskflow.logger.info(msg)
+    task.logger.info(msg)
+
+    # Now save cluster id in metadata
+    task.taskflow.set_metadata('cluster', cluster)
+
+    task.logger.info('Starting cluster.')
+
+    body = {
+        'status': 'launching'
+    }
+    client.patch('clusters/%s' % cluster['_id'], data=json.dumps(body))
+
+    secret_key = profile['secretAccessKey']
+    profile = profile.copy()
+    del profile['secretAccessKey']
+    log_write_url = '%s/clusters/%s/log' % (task.taskflow.girder_api_url,
+                                            cluster['_id'])
+    provision_params['cluster_state'] = 'running'
+    launch_params['cluster_state'] = 'running'
+    girder_token = task.taskflow.girder_token
+    cumulus.ansible.tasks.cluster.start_cluster(
+        launch_spec, provision_spec, cluster, profile, secret_key,
+        launch_params, provision_params, girder_token, log_write_url)
+
+    # Get the update to date cluster
+    cluster = client.get('clusters/%s' % cluster['_id'])
+
+    return cluster
+
+def create_girder_client(girder_api_url, girder_token):
+    client = GirderClient(apiUrl=girder_api_url)
+    client.token = girder_token
+
+    return client
+
+def _get_image(logger, profile, image_spec):
```

Comment:
This does not seemed to be used in either hpccloud or cumulus. Can it be removed?